apps/base-docs/tutorials/docs/0_gassless-transactions-with-paymaster.md

```diff
@@ -0,0 +1,616 @@
+import { privateKeyToSimpleSmartAccount } from 'permissionless/accounts';
+import { createPimlicoPaymasterClient } from 'permissionless/clients/pimlico';
+```
+
+### Setting up the constants for entrypoint and rpc url
+
+:::note Find your RPC URL
+
+The RPC URL is the URL for your Coinbase Developer Platform (CDP) Paymaster.
+
+This was saved in the previous section and follows this format: `https://api.developer.coinbase.com/rpc/v1/base/<SPECIAL-KEY>`
+
+Navigate to the [Paymaster Tool](https://portal.cdp.coinbase.com/products/bundler-and-paymaster) and select the `Configuration` tab at the top of the screen to obtain your RPC URL.
+
+:::
+
+We will create a constant for our RPC url obtained from cdp.portal.coinbase.com. The most secure way to do this is by using a proxy. For the purposes of this demo we will hardcode it into our `index.js` file. For product, we highly recommend using a [proxy service](https://www.smartwallet.dev/guides/paymasters).
+
+We will also need the address of the entrypoint contract for Base. A full list of entrpoint contracts and their addresses can be found [here](https://docs.alchemy.com/reference/factory-addresses).
+
+**Add the following to lines of code after your import statements:**
+
+```javascript
+const rpcUrl = https://api.developer.coinbase.com/rpc/v1/base/<SPECIAL-KEY>
+```
+
+```javascript
+const baseEntryPoint = '0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789';
+```
+
+```javascript
+const baseFactoryAddress = '0x15Ba39375ee2Ab563E8873C8390be6f2E2F50232';
+```
+
+We’ll start by creating a public client. Public clients enable our application to interact with the Ethereum blockchain. They use JSON-RPC API methods to perform actions such as retrieving block numbers, transactions, and reading from smart contracts. Learn more about this on viem.
+
+Initialize a public client named `publicClient` and set the chain to `base` and the transport to our `rpcURL` variable:
+
+```javascript
+const publicClient = createPublicClient({
+  chain: base,
+  transport: http(rpcUrl),
+});
+```
+
+### Create two smart accounts
+
+Now, we are going to create two SimpleAccounts for this demonstration. Collectively, these two accounts will help us test the security policies we set at both the global level and the per-account user operation level.
+
+To do so, we'll first need two private keys that will be used to create the Smart Accounts using Foundry. Open a terminal and run the following command **twice**:
+
+```bash
+cast wallet new
+```
+
+You should see something like this:
+
+```bash
+> cast wallet new
+Successfully created new keypair.
+Address: 0xD440D74620542...D6F005cfD9
+Private key: 0x01c9720c1dfa3c9...634793138897
+
+
+> cast wallet new
+Successfully created new keypair.
+Address: 0x5f8e5bC8620542...D6F005cfD9
+Private key: 0xbcd6fbc1dfa3c9...634793138897
+```
+
+Use the newly generated private keys as the parameter to the `privateKeyToSimpleSmartAccount` function and set the factory address to
```

Comment:
I'd add a danger admonishment that commiting these keys will give anyone access to these wallets.